### PR TITLE
0.2.1 Add String extension method compareToTr

### PR DESCRIPTION
**Pull request by uyarburak**
First of all, thanks for great library. 

The idea of exposing comparators is great but it is useful only if you have string list to sort.

I needed to sort a complex list based on a property value. Therefore, i needed compareTr function of yours which is private. In this PR, i have added a compareToTr extension method to String class.

### CODE REVIEW

File: lib/turkish.dart [hunk 1]
```diff
@@ -79,6 +83,10 @@ class _Turkish {
         : input.toLowerCase();
   }
 
+  int compareTr(String a, String b, bool ignoreCase) {
```
Comment (ahmetaa) [inline lib/turkish.dart:86]: Change the name of this method to `compareToTr` and private method `_compareTr` to `_compareToTr`. Private method name was not correct before.

File: README.md [hunk 2]
```diff
@@ -77,6 +87,13 @@ main() {
   print("Default= ${inputU.toLowerCase()}, "
       "Turkish=${turkish.toLowerCase(inputU)}\n");
 
+  // compareToTr
+  var zonguldak = "Zonguldak";
+  var cankiri = "Çankırı";
+  print("Comparison for [$zonguldak] and [$cankiri]");
+  print("Default= ${zonguldak.compareTo(cankiri)}, "
```
Comment (ahmetaa) [inline README.md:94]: Can you fix the print statements for all `=` symbols? Add a space before and after `=`.

**Review by ahmetaa — COMMENTED** [review]
Thanks, 
I think the reason comparator method was not exposed was that operator overloading would not work. However, this is still ok, as long as this method is used explicitly for comparison. 

Another way would be creating a new String type such as StringTr:String and overloading methods. But this would result in having casts all over.

**Comment by uyarburak** [comment]
I have done changes that you asked for in the second commit :)

By the way,
Extending String class is not possible as String class documentation states:  

> The `String` class cannot be extended or implemented. Attempting to do so yields a compile-time error.

**Review by ahmetaa — APPROVED** [review]
Lgtm, I will publish it to pub whenever possible.

**Pull request merged**